src/prefect/runner/runner.py

```diff
@@ -1399,12 +1399,12 @@ async def _submit_run_and_capture_errors(
             )
             status_code = process.returncode
         except Exception as exc:
-            if task_status:
+            if not task_status._future.done():
```

Comment:
The `BaseWorker` has this same check. Part of the issue in https://github.com/PrefectHQ/prefect/issues/17316 is that `task_status.started()` is erroneously called twice and this guards against that.